snuba/datasets/entities/sessions.py

```diff
@@ -79,6 +100,120 @@ def plus_columns(
+    neq(duration, Literal(None, MAX_UINT32)), eq(status, Literal(None, 1))
+)
+# `status IN (2,3,4)`
+terminal_status = in_condition(status, [Literal(None, status) for status in [2, 3, 4]])
+
+# These here are basically the same statements as the matview query
+sessions_raw_translators = TranslationMappers(
+    columns=[
+        ColumnToCurriedFunction(
+            None,
+            "duration_quantiles",
+            FunctionCall(None, "quantilesIf", quantiles),
+            (duration, duration_condition),
+        ),
+        ColumnToFunction(None, "duration_avg", "avgIf", (duration, duration_condition)),
+        ColumnToFunction(
+            None, "sessions", "sumIf", (quantity, eq(seq, Literal(None, 0))),
+        ),
+        ColumnToFunction(
+            None, "sessions_crashed", "sumIf", (quantity, eq(status, Literal(None, 2)))
+        ),
+        ColumnToFunction(
+            None, "sessions_abnormal", "sumIf", (quantity, eq(status, Literal(None, 3)))
+        ),
+        ColumnToFunction(
+            None,
+            "sessions_errored",
+            "plus",
+            (
+                FunctionCall(
+                    None,
+                    "uniqIf",
+                    (session_id, _and(has_errors, neq(session_id, lit_nil))),
+                ),
+                FunctionCall(
+                    None,
+                    "sumIf",
+                    (quantity, _and(terminal_status, eq(session_id, lit_nil))),
+                ),
+            ),
+        ),
+        ColumnToFunction(None, "users", "uniqIf", (distinct_id, did_not_nil)),
+        ColumnToFunction(
+            None,
+            "users_crashed",
+            "uniqIf",
+            (distinct_id, _and(eq(status, Literal(None, 2)), did_not_nil)),
+        ),
+        ColumnToFunction(
+            None,
+            "users_abnormal",
+            "uniqIf",
+            (distinct_id, _and(eq(status, Literal(None, 3)), did_not_nil)),
+        ),
+        ColumnToFunction(
+            None,
+            "users_errored",
+            "uniqIf",
+            (distinct_id, _and(has_errors, did_not_nil)),
+        ),
+    ]
+)
+
+
+class SessionsQueryStorageSelector(QueryStorageSelector):
+    def __init__(self) -> None:
+        self.materialized_storage = get_storage(StorageKey.SESSIONS_HOURLY)
+        self.raw_storage = get_storage(StorageKey.SESSIONS_RAW)
+
+    def select_storage(
+        self, query: Query, request_settings: RequestSettings
+    ) -> StorageAndMappers:
+        # time_range = get_time_range(query, "started")
+        # print(time_range)
+
+        granularity = extract_granularity_from_query(query, "started") or 3600
+        use_materialized_storage = granularity >= 3600
+
+        if use_materialized_storage:
+            return StorageAndMappers(
+                self.materialized_storage, sessions_hourly_translators
+            )
+        else:
+            return StorageAndMappers(self.raw_storage, sessions_raw_translators)
```

Comment:
Let's hide this with a feature flag at first. If there is an issue this will send a massive amount of load to the raw table upon deploy.
This is an example:
https://github.com/getsentry/snuba/blob/master/snuba/datasets/plans/single_storage.py#L58

Please default it to 0. Then we can turn it on dynamically in prod after deployment. If something is wrong we revert immediately without deployment.

Also please add a metric about which path you would be taking even when the feature flag is off. This is the best validation, if we see queries that would go to the raw table in the first few minutes after deployment we know something is wrong.

Here is an example for tracking metrics:
https://github.com/getsentry/snuba/blob/master/snuba/web/split.py#L220
(You can either use two different metric names (raw and aggregate) or one metric name and provide two different values for a tag.